### DISCUSSION TITLE
Is it possible to get code coverage while running multiple instances?

### DISCUSSION THREAD

**Discussion by FTSTimSin** [webdriverio/webdriverio#8947] [Q&A]
I'm using the code coverage capture shown here: https://webdriver.io/docs/devtools-service/#capture-code-coverage , but it doesn't seem to capture full coverage when I'm running tests concurrently. 

When I read the coverage report, it only appears to get the coverage of the last test file run. Is there any way to get the full coverage report of all the test files combined?

**Comment by christian-bromann** [comment]
> When I read the coverage report, it only appears to get the coverage of the last test file run. Is there any way to get the full coverage report of all the test files combined?

Is it possible that you write your coverage report to the same file with multiple instances?

**Reply by FTSTimSin** [reply]
Is there any way to get separate coverage reports for each test? This is how I set up my services in the config file:
```
services: [
    ['devtools', {
        coverageReporter: {
            enable: true,
            type: 'html', // html, lcov, json, text
            logDir: __dirname + '/coverage',
            exclude: [/vendors/, /modules.js/]
        }
    }],
  ],
```

**Comment by christian-bromann** [comment]
> When I read the coverage report

Which report are you talking about? As far as I can see WebdriverIO create coverage reports for every file.

**Reply by FTSTimSin** [reply]
I have multiple test files that all test the same index.js file. It's multiple test files, but the index.js.html coverage report gets overwritten each time a test file finishes I think.

I'm asking if I can get a separate index.js coverage reports for each test file.

**Reply by christian-bromann** [reply]
I think then it makes sense to include the browser id when [generating the file name](https://github.com/webdriverio/webdriverio/blob/main/packages/wdio-devtools-service/src/gatherer/coverage.ts#L111). Feel free to raise a PR for this.
